wdl/Module04b.wdl

```diff
@@ -558,9 +559,32 @@ task GetMedianSubset{
     max_retries: 1
   }
   RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
-  command {
-    R -e 'd<-read.table("~{medians}",header=T,check.names=F);result<-d[((apply(as.matrix(d[,5:length(d)]),FUN=median,1)<0.97 & apply(as.matrix(d[,5:length(d)]),FUN=median,1)>0.85)|(apply(as.matrix(d[,5:length(d)]),FUN=median,1)>1.03 & apply(as.matrix(d[,5:length(d)]),FUN=median,1)<1.65)),1:4];write.table(result,file="~{batch}_to_regeno.bed",sep="\t",quote=F,row.names=F,col.names=F)'
-  }
+  command <<<
+    set -euo pipefail
+    python3 <<CODE
+
+    from statistics import median
+    with open("~{medians}", 'r') as inp, open("~{batch}_to_regeno.bed", 'w') as outp:
+      for line in inp:
+        fields = line.strip().split('\t')
+        # first 4 fields are variant info (chr, start, end, varID)
+        var_info = fields[0:4]
+        # next len-4 fields are sample coverage medians for each sample in the batch
+        try:
+          coverage_medians = [float(x) for x in fields[4:]]
+        except ValueError as e:
+          if str(e) == "could not convert string to float: '.'":
```

Comment:
Matching an error message to a string like this is risky in case it changes with an update to the statistics package or one of its dependencies. It would be better to check for the presence of missing data explicitly:
```
if any([x == '.' for x in fields[4:]):
  print("Empty coverage data for variant %s at %s:%s-%s" % (var_info[3], var_info[0], var_info[1], var_info[2]))
  continue
else:
  med = median([float(x) for x in fields[4:]])
  ...
```